app/main/views/suppliers.py

```diff
@@ -408,13 +410,22 @@ def update_supplier_framework_details(supplier_id, framework_slug):
     if not interest_record:
         abort(404, "supplier_id '{}' has not registered interest in {}".format(supplier_id, framework_slug))
 
+    uniform_now = datetime.utcnow()
+
     if 'onFramework' in update_json:
         interest_record.on_framework = update_json['onFramework']
     if 'agreementReturned' in update_json:
         if update_json['agreementReturned']:
-            interest_record.agreement_returned_at = datetime.utcnow()
+            interest_record.agreement_returned_at = uniform_now
         else:
             interest_record.agreement_returned_at = None
+    if 'countersigned' in update_json:
+        if update_json['countersigned']:
+            interest_record.countersigned_at = uniform_now
+        else:
```

Comment:
Fair enough. Remove all the things and cut down on complex unnecessary behaviour.